File: Carta.java
class Carta{
    private String nome;
    private String img;
    private String categoria;

    public Carta(String n, String i, String c){
        this.nome=n;
        this.img=i;
        this.categoria=c;
    }

    public String getNome(){
        return this.nome;
    }

    public String getImg(){
        return this.img;
    }

    public String getCategoria(){
        return this.categoria;
    }

    public boolean corrisponde(Carta c){
        boolean si=false;
        if(this.nome.equals(c.getNome())){
            if(this.img.equals(c.getImg())){
                if(this.categoria.equals(c.getCategoria())){
                    si=true;
                }
            }
        }
        return si;
    }
}

File: PockemonTCGp_Grafica/PockemonTCGp_Grafica/Carta.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PockemonTCGp_Grafica
{
    public class Carta
    {
        private string Nome { get; set; }
        private string Immagine { get; set; }
        private string TipoCarta { get; set; }

        public Carta(string nome, string immagine, string tipoCarta)
        {
            this.Nome = nome;
            this.Immagine = immagine;
            this.TipoCarta = tipoCarta;
        }

        public string GetNome()
        {
            return this.Nome;
        }

        public string GetImg()
        {
            return this.Immagine;
        }

        public string GetCategoria()
        {
            return this.TipoCarta;
        }
    }
} //---Dzyubanov---
